### PR TITLE
Cleanups, docs

### PR DESCRIPTION
**Pull request by wkal-pubnub**

### CODE REVIEW

**Review by marcin-cebo — APPROVED** [review]

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Event.kt [hunk 1]
```diff
@@ -2,10 +2,34 @@ package com.pubnub.chat
...
+     */
     val chat: Chat
+
+    /**
+     * Timetoken of the message that triggered an event.
+     */
     val timetoken: Long
+
+    /**
+     * Data passed in an event (of [EventContent] subtype) that differ depending on the emitted event type ([EventContent.Typing]], [EventContent.Report], [EventContent.Receipt], [EventContent.Mention], [EventContent.Invite], [EventContent.Custom], [EventContent.Moderation], or [EventContent.TextMessageContent]).
```
[the remark this answers is not in the document]
Comment (wkal-pubnub) [reply pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Event.kt:22]: done

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Membership.kt [hunk 2]
```diff
@@ -5,27 +5,91 @@ import com.pubnub.kmp.CustomObject
 import com.pubnub.kmp.PNFuture
```
[the remark this answers is not in the document]
Comment (wkal-pubnub) [reply pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Membership.kt:6]: done

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Membership.kt [hunk 3]
```diff
@@ -5,27 +5,91 @@ import com.pubnub.kmp.CustomObject
...
+     *
+     * @param callback Defines the custom behavior to be executed when detecting membership changes.
+     * @return An [AutoCloseable] that you can use to stop receiving objects events by invoking [AutoCloseable.close].
+     */
     fun streamUpdates(callback: (membership: Membership?) -> Unit): AutoCloseable
 
+    /**
+     * Get a new [Membership] object updated with the values received in the [update].
+     *
+     * @return A [Membership] updated with values that are present in the [update]
```
[the remark this answers is not in the document]
Comment (wkal-pubnub) [reply pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Membership.kt:91]: done

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/config/CustomPayloads.kt [hunk 4]
```diff
@@ -4,20 +4,52 @@ import com.pubnub.api.JsonElement
 import com.pubnub.chat.types.EventContent
 
 class CustomPayloads(
+    /**
+     * Function that lets Chat SDK send your custom payload structure.
+     *
+     * The function will take an [EventContent.TextMessageContent] object and channel id as input,
```
[the remark this answers is not in the document]
Comment (wkal-pubnub) [reply pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/config/CustomPayloads.kt:10]: yeah, it's a `val` it doesn't have params XD 
I don't know if there's a way to add `@param` to params of a function type, but I don't think so?

I described the function though

File: pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/config/PushNotificationsConfig.kt [hunk 5]
```diff
@@ -3,10 +3,60 @@ package com.pubnub.chat.config
...
 class PushNotificationsConfig(
+    /**
+     * The main option for enabling sending notifications. It must be set to true if you want a particular client
+     * (whether a mobile device, web browser, or server) to send push notifications to mobile devices.
+     *
+     * These push notifications are messages with a provider-specific payload that the Chat SDK automatically attaches
+     * to every message. Chat SDK includes a default payload setup for [deviceGateway] in every message sent to the
+     * registered channels.
+     *
+     * This is the only required option to enable if you want to send push notifications to Android devices.
```
[the remark this answers is not in the document]
Comment (wkal-pubnub) [reply pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/config/PushNotificationsConfig.kt:19]: if you only want to send then that's the only one you need

**Pull request merged**